src/main/java/org/hiero/sketch/dataset/LocalDataSet.java

```diff
@@ -0,0 +1,47 @@
+package org.hiero.sketch.dataset;
+
+import org.hiero.sketch.dataset.api.*;
+import rx.Observable;
+
+public class LocalDataSet<T> implements IDataSet<T> {
+    private final T data;
+    // TODO: run on a separate thread
+
+    public LocalDataSet(final T data) {
+        this.data = data;
+    }
+
+    @Override
+    public <S> Observable<PartialResult<IDataSet<S>>> map(final IMap<T, S> mapper) {
+        final PartialResult<S> initial = new PartialResult<S>(0.0, null);
+        final Observable<PartialResult<S>> start = Observable.just(initial);
+        final Observable<PartialResult<S>> trsf = mapper.apply(this.data);
+        final Observable<PartialResult<S>> chain = start.concatWith(trsf);
+        final Observable<PartialResult<IDataSet<S>>> progress =
+                chain.map(e -> new PartialResult<IDataSet<S>>(e.deltaDone, null));
+        final PartialResultMonoid<S> mono = new PartialResultMonoid<S>(new NullMonoid<S>());
+        final Observable<PartialResult<S>> last = chain.reduce(mono.zero(), mono::add);
+        return progress.concatWith(last.map(e ->
+                new PartialResult<IDataSet<S>>(0.0, new LocalDataSet<S>(e.deltaValue))));
+    }
+
+    @Override
+    public <S> Observable<PartialResult<IDataSet<Pair<T, S>>>> zip(final IDataSet<S> other) {
+        if (!(other instanceof LocalDataSet<?>))
```

Comment:
There are multiple instances of this throughout the patch, but please use braces for if statements even if its just a single statement underneath. Makes the code easier to read.